File: Celebratorica/app/src/main/java/io/blushine/rmw/item/CategoryEvent.java
package io.blushine.rmw.item;

import java.util.List;

import io.blushine.android.common.ObjectEvent;

/**
 * Category event
 */
class CategoryEvent extends ObjectEvent<Category> {

CategoryEvent(Actions action) {
	super(action);
}

CategoryEvent(Actions action, Category category) {
	super(action, category);
}

CategoryEvent(Actions action, List<Category> categories) {
	super(action, categories);
}
}

File: Celebratorica/app/src/main/java/io/blushine/rmw/item/CategoryEvent.kt
package io.blushine.rmw.item

import io.blushine.android.common.ObjectEvent

/**
 * Category event
 */
internal class CategoryEvent @JvmOverloads constructor(action: Actions, items: List<Category> = listOf()) : ObjectEvent<Category>(action, items) {
	constructor(action: ObjectEvent.Actions, category: Category) : this(action, listOf(category))
}
